src/cogitate.py

```diff
@@ -1,58 +1,230 @@
-"""Command Line Interface for interacting with Github repository info."""
+"""Command Line Interface for the Cogitate tool."""
 
-# from data_collection import collect_commits
 import argparse
-
-# from pprint import pprint
-
-# from driller import find_repositories
-
+import os
+import validators
 import data_collection
+import data_processor
 import json_handler
 
 
-def main():
+def main(args):
     """Execute the CLI."""
-    args = retrieve_arguments()
-
-    # Currently only validates the PyGithub repository
+    if args["testwithprintargs"] == "y":
+        for key, value in args.items():
+            print(key, ":", value)
+        return
+    else:
+        pass
+    # Assess PyGithub access through token and repo path
     repository = data_collection.authenticate_repository(args["token"], args["repo"])
-
-    # Temporary structure given issue retrieval is the only function
-    contributor_data = data_collection.initialize_contributor_data(
-        "contributor_data_template"
-    )
-    contributor_data = data_collection.retrieve_issue_data(
-        repository, args["state"], contributor_data
-    )
-
-    # Intermediate between data_collection and data_processor
-    json_handler.write_dict_to_json_file(contributor_data, "contributor_data")
+    # Assess PyDriller access with link validator method
+    if repository == "INVALID" or link_validator(args["link"]) is False:
+        print("Cannot authenticate repository.")
+        return
+    else:
+        # Populate json file
+        data_collection.collect_and_add_raw_data_to_json(
+            args["link"], "raw_data_storage"
+        )
+        # calculate metrics to be used for team evaluation
+        issue_dict = {}
+        issue_dict = data_collection.retrieve_issue_data(
+            repository, args["state"], issue_dict
+        )
+        individual_metrics_dict = data_collection.calculate_individual_metrics()
+        merged_dict = data_collection.merge_metric_and_issue_dicts(
+            individual_metrics_dict, issue_dict
+        )
+        updated_dict = data_processor.add_new_metrics(merged_dict)
```

Comment:
This line should be after the data is merged correctly because some of these metrics are not designed to be merged such as the ratio.

Reply:
@noorbuchi  which data dictionary should be fed to it, metric or issue?

Reply:
nevermind I figured it out xD 
'add_new_metrics', definitely adding to the 'individual_metrics_dict.'

Reply:
Sorry for not clarifying, I just checked out your changes. The order the functions are expected to be called is as follows.

1. Collect and add raw data
2. Calculate indiv. metrics (can be switched with 3)
3. Calculate issue data, including authenticating repo (can be switched with 2)
4. Merge the two data sets
5. Prompt user to merge duplicate usernames
6. Add new metrics
7. Write the dictionary to JSON file, it now has, issue data, indiv. metrics, merged duplicates, and correctly calculated metrics

As of right now the only issue us that you have step 5 at the very last of the data collection process. You should try it to be like it is shown above. Please let me know if this clarifies it better or if you have other questions.

Reply:
when you have a chance @noorbuchi , I have updated Cogitate.py to follow the above listing of steps in the program, let me know if I mixed one up somewhere.

Reply:
Thanks for the update, I will check it out in a little bit since I'm finishing up some stuff in the web interface.